### DISCUSSION TITLE
Is there a way to use Kotlin as implenetation language of a plugin?

### DISCUSSION THREAD

**Discussion by H4kt** [tauri-apps/tauri#7336] [Q&A] [answered]
Hi,
I recently discovered Tauri as an Electron alternative (cuz npm/node sucks).
I'm very new to Rust language and very familiar with Kotlin.
I've seen a blog post about being able to use Kotlin for mobile development using Tauri, however I am currently building a desktop app.
Is there any way I can use Kotlin (or Kotlin/Native) as the implementation language of a Tauri plugin on desktop?

**Comment by FabianLars** [comment, marked as answer]
Not directly, no. The reason why it works for android plugins is solely because android apps _are_ java/kotlin.

That said, you _may_ be able to DIY it using ffi crates like https://github.com/astonbitecode/j4rs - of course that still requires quite a lot of Rust and idk how to make it include the JVM. -> you can also use Kotlin/Native (Kotlin with JVM probably too) as a sidecar but that only works in tauri apps and not plugins.

Other than that this has to wait for proper language binding support in tauri (will still take a _long_ time, but it's on the roadmap at least)

**Reply by Pseudow** [reply]
Hi, I have seen that Tauri was opened for JavaScript, since Kotlin can also be compiled in JavaScript wouldn't it be good to make support for Kotlin developers as well?

**Reply by H4kt** [reply]
> Hi, I have seen that Tauri was opened for JavaScript, since Kotlin can also be compiled in JavaScript wouldn't it be good to make support for Kotlin developers as well?

I don't think Kotlin/JS is the best target. It's preferably Kotlin/JVM or Kotlin/Native.

**Reply by FabianLars** [reply]
Kotlin/js would only work for frontend code, if you want to replace the rust backend (to a degree at least) then it'd be jvm or native indeed.

Kotlin/js is not something we really have to add support for on the Tauri side since Tauri is frontend agnostic anyway -> in other words, it should work already.

**Comment by kilisio** [comment]
@H4kt can you please direct me to the blog you stated in your question

**Reply by H4kt** [reply]
https://tauri.app/blog/2023/03/20/tauri-2-0-0-alpha-4/

**Reply by kilisio** [reply]
Thanks
